test/jdk/java/lang/StrictMath/Log1pTests.java

```diff
@@ -195,15 +217,71 @@ static int testLog1p() {
         return failures;
     }
 
-    public static void main(String [] argv) {
+    // Initialize shared random number generator
+    private static java.util.Random random = RandomFactory.getRandom();
+
+    /**
+     * Test StrictMath.log1p against transliteration port of log1p.
+     */
+    private static int testAgainstTranslit() {
         int failures = 0;
+        double x;
 
-        failures += testLog1p();
+        // Test just above subnormal threshold...
+        x = Double.MIN_NORMAL;
+        failures += testRange(x, Math.ulp(x), 1000);
 
-        if (failures > 0) {
-            System.err.println("Testing log1p incurred "
-                               + failures + " failures.");
-            throw new RuntimeException();
+        // ... and just below subnormal threshold ...
+        x = Math.nextDown(Double.MIN_NORMAL);
+        failures += testRange(x, -Math.ulp(x), 1000);
+
+         double[] conditionalPoints = {
+              1.0,
+             -0x1.0p-29,
+              0x1.0p-29,
+
+             -0x1.0p-54,
+              0x1.0p-54,
+
```

Comment:
maybe add a comment like
`/* approximations around sqrt(2)/2 - 1 and around sqrt(2) - 1 */`

Reply:
(I mean, for the doubles starting at L.246)

Reply:
> maybe add a comment like `/* approximations around sqrt(2)/2 - 1 and around sqrt(2) - 1 */`

Good clarification.